src/ui/index.js

```diff
@@ -0,0 +1,82 @@
+import { render, Box } from "@terrastack/ink";
+// import store from "./store";
+import { Stack } from "./components/stack";
+import React from "react";
+import { Provider } from "react-redux";
+
+const { createStore } = require("redux");
+
+const store = createStore((state = {}, action) => {
+  if (action.component == undefined) {
+    return state;
+  }
+
+  let layer = action.component._layer;
+  let name = action.component.name;
+  let newState = {};
+  switch (action.type) {
+    case "ADDED":
+      if (state[layer] == undefined) {
+        state[layer] = {};
+      }
+      newState = Object.assign({}, state);
+      newState[layer][name] = "added";
+      return newState;
+    case "START":
+      if (state[layer] == undefined) {
+        state[layer] = {};
+      }
+      newState = Object.assign({}, state);
+      newState[layer][name] = "start";
+      return newState;
+    case "SUCCESS":
+      newState = Object.assign({}, state);
+      newState[layer][name] = "success";
+      return newState;
+    case "FAILED":
+      newState = Object.assign({}, state);
+      newState[layer][name] = "failed";
+      return newState;
+    default:
+      return state;
+  }
+});
+
+const applyVisualization = base => {
+  let buffer = [];
+
```

Comment:
Maybe we can publish the initial state here? Or somehow fill it in the state directly?

IMHO: It make sense to move the initial state publishing to the clinic, as it is not needed anywhere else..